components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/service/APIKeyValidationService.java

```diff
@@ -149,12 +168,26 @@ public APIKeyValidationInfoDTO validateKey(String context, String version, Strin
         validationContext.setValidationInfoDTO(new APIKeyValidationInfoDTO());
         validationContext.setVersion(version);
 
+        if (Util.tracingEnabled()) {
+            getAccessTokenCacheSpan =
+                    Util.startSpan(TracingConstants.GET_ACCESS_TOKEN_CACHE_KEY, validateMainSpan, tracer);
+        }
         String cacheKey = APIUtil.getAccessTokenCacheKey(accessToken,
                                                          context, version, matchingResource, httpVerb, requiredAuthenticationLevel);
 
         validationContext.setCacheKey(cacheKey);
+        if (Util.tracingEnabled()) {
+            Util.finishSpan(getAccessTokenCacheSpan);
+        }
 
+        if (Util.tracingEnabled()) {

```

Comment:
Why do we have multiple if statements which checks for the same condition?

Reply:
Tracing should be done only if it is enabled. That is why I'm starting and finishing the span with if conditions. If I use only one "IF" condition, the in-between code won't be executed if tracing is not enabled